### PR TITLE
Add thread based strand scheduling

### LINKED ISSUE #19456: [Spec Deviation] Strands created from named-worker-decls are scheduled in different threads
In the absence of annotations, the runtime scheduler should schedule strands created from named-worker-decls in the same thread. 

Here is a sample code to reproduce this in jBallerina.

```ballerina
import ballerina/io;
import ballerinax/java;

public function main() {
    io:println("default: " + getCurrentThreadName());
    worker w1 {
        io:println("w1: " + getCurrentThreadName());
    }

    worker w2 {
        io:println("w2: " + getCurrentThreadName());
    }
}

function getCurrentThreadName() returns string {
    handle t = currentThread();
    handle tName = getName(t);
    return java:toString(tName) ?: "";
}

function currentThread() returns handle = @java:Method {
    class: "java.lang.Thread"
} external;

function getName(handle thread) returns handle = @java:Method {
    class: "java.lang.Thread"
} external;
```

### PR DESCRIPTION
**Pull request by vinok88**
In the current implementation, thread selection for a concurrent strand is round-robin.

Fixes #19456

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by Kishanthan** [comment]
any performance impact for the default scheduler ?

**Comment by vinok88** [comment]
Following are the results of the test I ran locally,

Counting rows of 4 csv files with 27000 rows.

|  Scenario |  New Scheduler |  Scenario |
|---|---| ---|
| Sequentially  |  953603ms | 1062881ms |
| Sequentially but with 4workers | 979031ms | -- |
|Concurrent 4 workers | 306206ms | 320661ms |

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/AnnotationUtils.java [hunk 1]
```diff
@@ -92,4 +93,16 @@ public static void processFPValueAnnotations(FPValue fpValue, MapValue globalAnn
...
         }
     }
+
+    /**
+     * Returns true if given {@link FPValue} is annotated to be run concurrently.
+     * @param fpValue function pointer to be invoked
+     * @return true if should run concurrently
+     */
+    public static boolean isConcurrent(FPValue fpValue) {
+        if (((BFunctionType) fpValue.getType()).getAnnotation("ballerina/lang.annotations", "concurrent") != null) {
```
Comment (lankavitharana) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/AnnotationUtils.java]: can simplify this to a single line

Comment (vinok88) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/AnnotationUtils.java]: Fixed. Thanks.

**Comment by vinok88** [comment]
This will be addressed in https://github.com/ballerina-platform/ballerina-lang/pull/20115 with a new design.

**Pull request closed without merging**